### PR TITLE
Adding release script and updating submodules

### PR DESCRIPTION
**Pull request by ppathan**
# Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Checklist:

- [x] All new and existing tests passed.
- [x] The code follows the code style and conventions of this project.
- [ ] The change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

# Description of the change

* Adding release script to automate the release process
* Updating the reverse-proxy build script
* Updating the sub-modules

# Motivation for the change

* Making the release process easier and less prone to manual errors

### CODE REVIEW

File: reverse-proxy/build [hunk 1]
```diff
@@ -3,8 +3,28 @@
...
 DOCKER_IMAGE="azureiotpcs/remote-monitoring-nginx"
 
-DOCKER_LABEL2="Commit=$(git log --pretty=format:'%H' -n 1)"
-DOCKER_LABEL3="Date=$(/usr/bin/env date +%Y-%m-%dT%H:%M:%S)"
+APP_HOME="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && cd .. && pwd )/"
 
-docker build --compress --tag "$DOCKER_IMAGE:testing" \
-  --label "$DOCKER_LABEL2" --label "$DOCKER_LABEL3" .
+build_docker() {
+  DOCKER_LABEL2="Commit=$(git log --pretty=format:'%H' -n 1)"
```
Comment (dluc) [inline reverse-proxy/build]: nit: 4 spaces indent

Comment (ppathan) [reply reverse-proxy/build]: Fixed.

File: scripts/git/release.sh [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+
+failed() {
+    SUB_MODULE=$1
+    echo -e "${RED}Cannot find directory $SUB_MODULE${NC}"
+    exit 1
+}
+
+check_input() {
+    if [ ! -n "$VERSION" ]; then
+        echo -e "${RED}Version is required parameter${NC}"
```
Comment (dluc) [inline scripts/git/release.sh]: nit: "...is *a* required..."

File: scripts/git/release.sh [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+        exit 1
+    elif [ ! -n "$ACCESS_TOKEN" ]; then
+        echo -e "${RED}Acess_token is required parameter${NC}"
```
Comment (dluc) [inline scripts/git/release.sh]: typo

File: scripts/git/release.sh [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+
+    echo
+    echo -e "${CYAN}====================================     End: Tagging $REPO_NAME repo     ====================================${NC}"
+    echo
+
+    echo
+    echo -e "${CYAN}====================================     Start: Release for $REPO_NAME     ====================================${NC}"
+    echo
+
+    DATA="{
```
Comment (dluc) [inline scripts/git/release.sh]: please add link to doc explaining this data record

File: scripts/git/release.sh [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+        \"tag_name\": \"$VERSION\",
+        \"target_commitish\": \"master\",
+        \"name\": \"$VERSION\",
+        \"body\": \"$DESCRIPTION\",
```
Comment (dluc) [inline scripts/git/release.sh]: we might want some validation to ensure $DESCRIPTION doesn't break the json format

Comment (ppathan) [reply scripts/git/release.sh]: The description is a string that is in quotes. If somebody passes a json that will be treated as a string. Long term the goal is for this to be a path to RELEASE.md

**Pull request merged**